sql/schema_changer.go

```diff
@@ -62,3 +68,81 @@ func (p *planner) applyUpVersion(tableDesc *TableDescriptor) {
 		tableDesc.Version++
 	}
 }
+
+// SchemaChanger is used to change the schema on a table.
+type SchemaChanger struct {
+	tableID ID
+	node    roachpb.NodeID
+	txn     *client.Txn
+}
+
+// NewSchemaChangerForTesting only for tests.
+func NewSchemaChangerForTesting(tableID ID, node roachpb.NodeID, txn *client.Txn) SchemaChanger {
```

Comment:
How will a SchemaChanger be created outside of tests? The design seems strange to me: we won't have multiple lease operations within a single transaction, will we? I think the `*client.Txn` should be an argument to the lease methods instead of a field of the `SchemaChanger` (and then what's the point of the SchemaChanger type vs a bunch of static functions?)